websockets/websockets_test.go

```diff
@@ -879,3 +880,241 @@ func TestCookiesDefaultJar(t *testing.T) {
+			if (data != "` + text + `"){
+				throw new Error("wrong message received from server: ", data)
+			}
+
+			ws.close()
+		}
+		`))
+			return runErr
+		})
+
+		require.NoError(t, err)
+
+		samples := metrics.GetBufferedSamples(ts.samples)
+		assertSessionMetricsEmitted(t, samples, "", sr("WSBIN_URL/ws-compression"), http.StatusSwitchingProtocols, "")
+	})
+
+	t.Run("params", func(t *testing.T) {
+		t.Parallel()
+		testCases := []struct {
+			compression   string
+			expectedError string
+		}{
+			{
+				compression:   `""`,
+				expectedError: `unsupported compression algorithm '', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `null`,
+				expectedError: `unsupported compression algorithm 'null', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `undefined`,
+				expectedError: `unsupported compression algorithm 'undefined', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `" "`,
+				expectedError: `unsupported compression algorithm '', supported algorithm is 'deflate'`,
+			},
+			{compression: `"deflate"`},
+			{compression: `"deflate "`},
+			{
+				compression:   `"gzip"`,
+				expectedError: `unsupported compression algorithm 'gzip', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `"deflate, gzip"`,
+				expectedError: `unsupported compression algorithm 'deflate, gzip', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `"deflate, deflate"`,
+				expectedError: `unsupported compression algorithm 'deflate, deflate', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `"deflate, "`,
+				expectedError: `unsupported compression algorithm 'deflate,', supported algorithm is 'deflate'`,
+			},
+		}
+
+		for _, testCase := range testCases {
+			testCase := testCase
+			t.Run(testCase.compression, func(t *testing.T) {
+				t.Parallel()
+				ts := newTestState(t)
+				sr := ts.tb.Replacer.Replace
+				ts.tb.Mux.HandleFunc("/ws-compression-param", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					upgrader := websocket.Upgrader{
+						EnableCompression: true,
+						ReadBufferSize:    1024,
+						WriteBufferSize:   1024,
+					}
+
+					conn, e := upgrader.Upgrade(w, req, w.Header())
+					if e != nil {
+						t.Fatalf("/ws-compression-param cannot upgrade request: %v", e)
+						return
+					}
+
+					e = conn.Close()
+					if e != nil {
+						t.Logf("error while closing connection in /ws-compression-param: %v", e)
+						return
+					}
+				}))
+
+				err := ts.ev.Start(func() error {
+					_, runErr := ts.rt.RunString(sr(`
+					var ws = new WebSocket("WSBIN_URL/ws-compression-param", null, {"compression":` + testCase.compression + `})
+					ws.onopen = () => {
+						ws.close()
+					}
+					`))
+					return runErr
+				})
+
+				if testCase.expectedError == "" {
+					require.NoError(t, err)
+				} else {
+					require.Error(t, err)
+					require.Contains(t, err.Error(), testCase.expectedError)
```

Comment:
```suggestion
					require.ErrorContains(t, err, testCase.expectedError)
```

https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorContains